tutorial-compiler/index.asciidoc

```diff
@@ -0,0 +1,117 @@
+
+Afterwards we will show you, how to write your own tutorials.
+
+Requirements: 
+ *  GitHub account
+
+You can find the documentation here:
+
+https://github.com/devonfw-forge/tutorial-compiler/wiki
+
+====
+
+====
+# Setup your workspace
+Let's start!
+
+First of all, go to the tutorial-compiler repository on  GitHub:
+
+https://github.com/devonfw-tutorials/tutorial-compiler
+
+[step]
+--
+nextKatacodaStep("Fork the 'tutorial-compiler'-repository", [{ "image": "images/fork.jpg" }])
+--
+
+Fork the repository by clicking on the fork-button.
+This will create a copy of this repository in your GitHub account. You will find it in 'Your repositories'.
+====
+
+====
+Go to your repositories, click on 'tutorial-compiler' and copy the repository URL as shown in image below.
+
+[step]
+--
+nextKatacodaStep("Copy the 'tutorial-compiler'-URL", [{ "image": "images/clone_code.jpg" }])
+--
+
+====
+
+====
+Usually you would paste the copied URLs next to the 'git clone' - command, but in this tutorial we will clone the original 'tutorial-compiler'-repository. Therefore we are able to write the asciidoc-files and generate the tutorials, but we can not push changes in the source code to it.
+[step]
+--
+cloneRepository("", "https://github.com/devonfw-tutorials/tutorial-compiler.git")
+--
+====
+
+====
+Switch to the 'tutorials'-repository and repeat all the previous steps:
+
+https://github.com/devonfw-tutorials/tutorials.git
+
+- Fork the 'tutorials'-repository
+- Copy the 'tutorials'-URL
+
+[step]
+--
+cloneRepository("", "https://github.com/devonfw-tutorials/tutorials.git")
+--
+After this step you should have both repositories in the same folder next to each other.
+====
+
+[step]
+--
+npmInstall("", {"name": "typescript", "global": true})
+--
+
+[step]
+--
+npmInstall("tutorial-compiler")
+--
+
+# Write the my-thai-star tutorial
+
+Let's start with the second part of this tutorial and write the "my-thai-star" tutorial.
+
+[step]
+--
+createFile("tutorials/my-thai-star/index.asciidoc")
+--
+
+Add the tutorial title "My Thai Star"
+[step]
+--
+changeFile("tutorials/my-thai-star/index.asciidoc", {"content": "= My Thai Star"})
+--
+
+You have to write the introduction in between these equal signs '====' and underneath the title. 
+
+[step]
+--
+changeFile("tutorials/my-thai-star/index.asciidoc", {"file": "text/intro.txt", "placeholder": "= My Thai Star" })
+--
+
+====
+Last but not least add all the needed functions for this tutorial.
+
+Follow the link below to find a list of all available functions and their descriptions
+https://github.com/devonfw-tutorials/tutorial-compiler/wiki/Functions
+
+Feel free to add your own functions to the compiler
+You will find all the information you need on:
+
+https://github.com/devonfw-tutorials/tutorial-compiler/wiki/Development
+[step]
+--
+changeFile("tutorials/my-thai-star/index.asciidoc", {"file": "text/functions.txt", "placeholder": "This is the online demo of https://github.com/devonfw/my-thai-star" })
+--
+
```

Comment:
We should also add a step that shows how to run the tutorial with the tutorial-compiler (either with executeFile or nextKatacodaStep)